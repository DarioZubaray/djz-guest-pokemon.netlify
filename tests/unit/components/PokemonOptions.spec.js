import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'
import { mockPokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions components', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallowMount( PokemonOptions, {
            props: {
              disabledButtons: false,
              pokemons: mockPokemons
            }
        })
    })

    test('debe de hacer match con el snapshot', () => {
      expect( wrapper.html() ).toMatchSnapshot()
    })

    test('debe de mostrar las opciones correctas', () => {

        const [ li1, li2, li3, li4] = wrapper.findAll('li')

        expect( li1.exists() ).toBeTruthy()
        expect( li1.text() ).toBe(mockPokemons[0].name.toUpperCase())
        expect( li2.exists() ).toBeTruthy()
        expect( li2.text() ).toBe(mockPokemons[1].name.toUpperCase())
        expect( li3.exists() ).toBeTruthy()
        expect( li3.text() ).toBe(mockPokemons[2].name.toUpperCase())
        expect( li4.exists() ).toBeTruthy()
        expect( li4.text() ).toBe(mockPokemons[3].name.toUpperCase())
    })

    test('debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
        const pokemonEmitted = {
            "id": 1,
            "name": "bulbasaur",
        }
        const [ li1, li2, li3, li4] = wrapper.findAll('li')

        li1.trigger('click')

        expect ( wrapper.emitted('selection').length ).toBe(1)
        expect ( wrapper.emitted('selection')[0] ).toEqual([pokemonEmitted])
    })

})