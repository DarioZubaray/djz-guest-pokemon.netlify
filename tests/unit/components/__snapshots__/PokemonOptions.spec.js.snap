// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions components debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li disabled="disabled">BULBASAUR</li>
    <li disabled="disabled">IVYSAUR</li>
    <li disabled="disabled">VENUSAUR</li>
    <li disabled="disabled">CHARMANDER</li>
  </ul>
</div>
`;
